File: BasicPlaybackSampleApp copy/BasicPlaybackSampleApp/ManifestClient.m
//
//  ManifestClient.m
//  BasicPlaybackSampleApp
//
//  Created by alex on 1/11/2016.
//  Copyright (c) 2016, Telstra.
//  This is proprietary information of the Telstra Corporation Limited.
//  Copying or reproduction without prior written approval is prohibited.
//


#import <Foundation/Foundation.h>
#import "ManifestClient.h"
#import "NSDictionary+manifestDigest.h"
#import "UIImageView+AFNetworking.h"
#import "AFHTTPRequestOperation.h"
#import "NSData+Base64.h"
#import "NetworkUtils.h"


static NSString * const BaseURLString = @"http://localhost:5000";


@interface ManifestClient ()
@property(strong) NSDictionary *manifestDigest;
@property(strong) NSArray *manifestDigests;

@end

@implementation ManifestClient

- (void) getManifest
{
    NSString* manifestUrl = @"http://player.ooyala.com/hls/playready/iphone/master/51cnc5eDotMHwAbScZXy2FSp_zXKinMh.m3u8";
    NSString *urlString = [NSString stringWithFormat:@"%@/services/command/v1/manifests?url=%@&includeUrls=true", BaseURLString, manifestUrl];
    
    AFHTTPRequestOperation *operation = [NetworkUtils getOperation:urlString];
   
    [operation setCompletionBlockWithSuccess:^(AFHTTPRequestOperation *operation, id responseObject) {
        self.manifestDigests = (NSArray *)responseObject;
        if ([self.manifestDigests count] > 0) {
            self.manifestDigest = [self.manifestDigests firstObject];
            NSDictionary* urls = [self.manifestDigest urls];
            NSDictionary* summary = [self.manifestDigest summary];
            if ([urls count] > 0) {
                for (NSString* singleUrl in urls.allKeys) {
              //      NSURL *url = [NSURL URLWithString:singleUrl];
                }
            }
        }
        
    } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
        self.manifestDigests = [[NSArray alloc] init];    //Error handling goes here.
    }];
    
 
    [operation start];
}


@end

File: BasicPlaybackSampleApp/BasicPlaybackSampleApp/ManifestClient.swift
//
//  ManifestClientSwift.swift
//  BasicPlaybackSampleApp
//
//  Created by admin on 4/11/16.
//  Copyright © 2016 Telstra. All rights reserved.
//

import Foundation

@objc class ManifestClient : NSObject {
    let BaseURLString = "http://localhost:5000";
    
    func getManifest(manifestUrl:String, includeUrls:Bool, includeSegments:Bool,
                     successResponse:@escaping ([ManifestDigest])->Void, failureResponse:@escaping (String)->Void)
    {
         let urlString = "\(BaseURLString)/services/command/v1/manifests?url=\(manifestUrl)&includeUrls=true"
        if let operation = NetworkUtils.getOperation(urlString) {
            
            operation.setCompletionBlockWithSuccess(
                { (operation: AFHTTPRequestOperation?, responseObject: Any?) in
                    if let error = responseObject as? NSDictionary {
                        if (error.count > 0) {
                           //TODO: unmarshall me to get error reason
                        }// can get status 401 here if password is invalid.
                        failureResponse("errorPayload")
                    } else if let manifestDigestsJson = responseObject as? NSArray {
                        do{
                            var manifestDigests = [ManifestDigest]()
                            for individual in manifestDigestsJson {
                                let manifestDigest = try ManifestDigest(object: individual as! JSONObject)
                                    manifestDigests.append( manifestDigest)
                            }
                            successResponse(manifestDigests)
                        } catch {
                                 //TODO:
                        }                        
                    }
                },
                failure: { (operation, error) in
                    if let errorDescription = error?.localizedDescription {
                        failureResponse(errorDescription)
                    } else {
                        failureResponse("unknown error");
                    }
                }
            )
            
            operation.start();
        }
    }
    
    // TODO: move me to ManifestService
    func getManifestHarness(){
        let manifestUrl = "http://player.ooyala.com/hls/playready/iphone/master/51cnc5eDotMHwAbScZXy2FSp_zXKinMh.m3u8"

        getManifest(manifestUrl: manifestUrl, includeUrls: true, includeSegments: true, successResponse: {manifestDigest in
            
            },
            failureResponse: {error in
        })
    }
    
}
